geode-core/src/main/java/org/apache/geode/internal/cache/BucketRegionQueue.java

```diff
@@ -453,21 +454,22 @@ public Object peek() {
    * If a matching object also fulfills the endPredicate then the method
    * stops looking for more matching objects.
    */
-  public List<Object> getElementsMatching(Predicate matchingPredicate, Predicate endPredicate) {
+  public List<Object> getElementsMatching(Predicate<GatewaySenderEventImpl> matchingPredicate,
```

Comment:
You might want to change from `GatewaySenderEventImpl` to your new `InternalGatewaySenderEvent` everywhere you've added new code. I don't see anything that would obviously break.